### PR TITLE
Add MC app failed for ludacris and remove outdated cluster-autoscaler rules

### PR DESCRIPTION
**Pull request by rossf7**
This PR:

- Add missing `ManagementClusterAppFailedLudacris` rule.
- Remove outdated `cluster-autoscaler` app rules now we have owners annotation in Chart.yaml https://intranet.giantswarm.io/docs/dev-and-releng/app-developer-processes/routing_app_alerts/#owners-annotation

### Checklist

- [x] Update changelog in CHANGELOG.md.

### CODE REVIEW

File: helm/prometheus-rules/templates/alerting-rules/app.rules.yml [hunk 1]
```diff
@@ -11,11 +11,11 @@ spec:
   groups:
   - name: app
     rules:
-    - alert: ManagementClusterAppFailedBatman
+    - alert: ManagementClusterAppFailedAtlas
```
Comment (rossf7) [inline helm/prometheus-rules/templates/alerting-rules/app.rules.yml:14]: Diff is a pain but this just adds `ManagementClusterAppFailedLudacris` and orders the other rules.

File: helm/prometheus-rules/templates/alerting-rules/app.rules.yml [hunk 2]
```diff
@@ -122,118 +121,25 @@ spec:
         cancel_if_outside_working_hours: "true"
         severity: page
         sig: none
-        team: batman
-        topic: releng
-    {{- if eq .Values.Installation.V1.Provider.Kind "aws" }}
```
Comment (rossf7) [inline helm/prometheus-rules/templates/alerting-rules/app.rules.yml:127]: These per provider rules are no longer needed.

**Comment by rossf7** [comment]
Tested on gauss.

**Review by tomahawk28 — APPROVED** [review]
Find for me, what do you think to ask team-firecracker for a review?

**Comment by rossf7** [comment]
> Find for me, what do you think to ask team-firecracker for a review?

This doesn't affect the routing of the alerts. It just simplifies the rules so they are easier to manage. So I'd like to go with this.

**Pull request merged**